src/main/java/core/basesyntax/dao/ma/MentorDaoImpl.java

```diff
@@ -11,6 +14,13 @@ public MentorDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public List<Mentor> findByAgeGreaterThan(int age) {
-        return null;
+        try (Session session = sessionFactory.openSession()) {
+            Query<Mentor> getAgeGreaterThanQuery = session.createQuery("SELECT m FROM Mentor m "
+                    + "WHERE m.age > :age", Mentor.class);
+            getAgeGreaterThanQuery.setParameter("age", LocalDate.now().getYear() - age);
```

Comment:
for mentor age it's the number of full years( 23 in my case). So here you can just compare two numbers.

Reply:
done